wp-includes/rest-api/endpoints/class-wp-rest-customize-changesets-controller.php

```diff
@@ -610,6 +747,13 @@ protected function prepare_item_for_database( $request ) {
 				}
 
 				if ( isset( $data[ $setting_id ] ) ) {
+
+					// If the value of the setting is null, this should be removed from the changeset. @todo confirm logic.
+					if ( null === $data[ $setting_id ] ) {
```

Comment:
Good question. Yes, I think that if the entire `params` is set to `null` that this should indicate that the setting is being deleted. So consider you have a changeset that contains:

```json
{
"blogname": { "value": "Title" },
"blogdescription": { "value": "Title" }
}
```

If the `PUT` (or `PATCH` request) contains:

```json
{
"blogname": null
}
```

Then this should result in the entire `blogname` from being removed from the changeset, while the `blogdescription` would be left as-is.

Alternatively, if someone submitted a `PATCH` request with:

```json
{
"blogname": { "value": null }
}
```

Then I think this should actually be an error and the request should be aborted. Customizer settings weren't intended to allow `null` to be stored, where `null` originally indicated a setting invalidity.

Lastly, if someone `PATCH`ed a request with:

```json
{
"blogname": { "custom": "abc" }
}
```

Then this could result in:

```json
{
"blogname": { "value": "Title", "custom": "abc" },
"blogdescription": { "value": "Title" }
}
```

So essentially this would allow for the `params` for a given setting to be a key-value store. This is currently supported when you call:

```js
wp.customize.requestChangesetUpdate({ blogname: { custom: 'abc' }})
```

If you inspect the changeset after doing such a call, you'll see:

```json
{
    "blogname": {
        "value": "Title",
        "type": "option",
        "user_id": 1,
        "custom": "abc"
    }
```

The `type` and `user_id` props should be read-only.